frontend/src/views/SecretMainPage/components/SecretListView/SecretReferenceDetails.tsx

```diff
@@ -0,0 +1,124 @@
+import { twMerge } from "tailwind-merge";
+
+import { FormControl, SecretInput, Tag, Tooltip } from "@app/components/v2";
+import { useWorkspace } from "@app/context";
+import { useGetSecretReferenceTree } from "@app/hooks/api";
+import { SecretV3RawSanitized, TSecretReferenceTraceNode } from "@app/hooks/api/types";
+
+type Props = {
+  environment: string;
+  secretPath: string;
+  secret: SecretV3RawSanitized;
+};
+
+export const SecretReferenceNode = ({
+  node,
+  isRoot
+}: {
+  node: TSecretReferenceTraceNode;
+  isRoot?: boolean;
+}) => (
+  <div>
+    {isRoot ? (
+      <FormControl label="Reference value" className="sticky top-0 mb-0 bg-mineshaft-800">
+        <SecretInput
+          key="value-overriden"
+          isReadOnly
+          value={node.value}
+          containerClassName="text-bunker-300 hover:border-primary-400/50 border border-mineshaft-600 bg-bunker-800  px-2 py-1.5"
+        />
+      </FormControl>
+    ) : (
+      <div className="py-2 text-sm">
+        <div>
+          {node.key}
+          <Tooltip content={node.value}>
+            <span
+              className={twMerge(
+                "ml-2 rounded border border-gray-400 px-1 text-xs text-gray-400",
+                !node.value && "border-red-400 text-red-400"
+              )}
+            >
+              {node.value ? "value" : "empty"}
+            </span>
```

Comment:
I think this should be a little more clear that if you hover it shows the value - maybe an eye icon or even `Show Value` or `Reveal Value` - it kind of looks like a badge to me right now - not something that a user would intuitively click/hover over

Reply:
Changed to Reveal Value